providers/rpc/doc.go

```diff
@@ -0,0 +1,8 @@
+/*
+Package rpc is a provider that defines an HTTP request/response contract for handling BMC interactions.
+It allows users a simple way to interoperate with an existing/bespoke out-of-band management solution.
+
+The rpc provider request/response payloads are modeled after JSON-RPC 2.0, but are not JSON-RPC 2.0
```

Comment:
Out of curiousity, 
 - Why pick JSON-RPC in the first place, instead of lets say https://github.com/twitchtv/twirp
 - If JSON-RPC seems ideal for this usecase, could we not use an existing library https://github.com/sourcegraph/jsonrpc2
 
I'm a bit wary of having to maintain our own JSON-RPC implementation.

Reply:
ah yeah, very fair question. Flexibility is the biggest factor. I have found being conformant to a spec as a barrier to interoperating with bespoke out-of-band systems. I shied away from any protocol buffers because of the user experience for creating clients and interacting with the provider. I did look into using https://github.com/sourcegraph/jsonrpc2. The biggest reason i did not go with it was I couldn't find a way to add headers to the calls. Having some sort of authn/authz was important, in my experience, and i just couldn't find a way to add auth to headers in this library. And secondary, it meant that we would be using the strict JSON-RPC spec.

Reply:
I agree i dont want to maintain our own JSON-RPC implementation. I think at the moment the contract we have is just a request/response and it's about as simple/basic as i could think. It's HTTP and its general constructs like body and headers.

Reply:
Thanks for the explanation Jacob, as long we keep the foot print small it should be maintainable. Are you able to include a consumer example for reference in the examples/ directory?